src/CMakeLists.txt

```diff
@@ -131,6 +131,36 @@ ign_build_tests(TYPE UNIT
     ignition-gazebo${PROJECT_VERSION_MAJOR}
 )
 
+if(TARGET UNIT_ign_TEST)
+
+  # Running `ign gazebo` on macOS has problems when run with /usr/bin/ruby
+  # due to System Integrity Protection (SIP). Try to find ruby from
+  # homebrew as a workaround.
```

Comment:
This feels a bit out of place here just for the tests. It feels to me that it should either be something that we set at a higher level, such as:

* when we find Ruby using `ign-cmake`
* when we find Ruby for `ign-tools`, and export it to all its dependencies

Or it should be set by the user / machine running the tests, outside of the test code.

My main concerns are:

* this place may be a bit hidden for someone trying to understand howcome the tests pass but their setup fails
* having to duplicate this logic for all similar command line tests
* having a different environment for runtime and for testing

Any thoughts?

Reply:
> This feels a bit out of place here just for the tests. It feels to me that it should either be something that we set at a higher level, such as:
> 
> * when we find Ruby using `ign-cmake`
> * when we find Ruby for `ign-tools`, and export it to all its dependencies

we don't actually find or export Ruby in `ign-tools`; whichever `ruby` is in the `PATH` at runtime will be used

the only place I know of where we find Ruby during build time is sdformat, but that's just to generate xml schema files, and it doesn't become a run-time dependency (except via `ign-tools`, but that is similar to our other packages)

> Or it should be set by the user / machine running the tests, outside of the test code.

this would require additional configuration of our CI machines / build scripts. I thought it would be simpler to give cmake a hint about which ruby to prefer

> * this place may be a bit hidden for someone trying to understand howcome the tests pass but their setup fails

I've added a console error message to `ign gazebo` specifically for macOS when there are problems that seem related to SIP that recommends using the brew version of ruby to mitigate this

> * having to duplicate this logic for all similar command line tests

in most of our other packages, the `ign.cc` files are linked directly into the main package library, but here it is linked into a component library. I believe SIP causes problems with the RPATH relating the component library to the main project library when trying to dlopen, which is why it is an issue here (and in https://github.com/ignitionrobotics/ign-plugin/pull/32 which uses the loader component) but not in ign-transport or ign-msgs. I don't know exactly how SIP works, but this is my best guess

> * having a different environment for runtime and for testing

there is a console message recommending the brew ruby if there are errors, which should encourage people to use a setup that matches our test environment

Reply:
> we don't actually find or export Ruby in ign-tools; whichever ruby is in the PATH at runtime will be used

Sure, but what I'm wondering is whether this is the best behaviour for users of `ign-tools`.

> in most of our other packages, the ign.cc files are linked directly into the main package library, but here it is linked into a component library.

I see, so the fix isn't needed for all libraries, so it probably makes sense to do it only for the libraries that need it, instead of forcing everything that depends on `ign-tools` to use ruby from brew.

---

Thanks for the clarifications, I think it's ok like this.